scrapy/spiders/__init__.py

```diff
@@ -42,9 +43,9 @@ def __init__(self, name: Optional[str] = None, **kwargs: Any):
             self.start_urls: List[str] = []
 
     @property
-    def logger(self) -> logging.LoggerAdapter:
+    def logger(self) -> SpiderLoggerAdapter:
         logger = logging.getLogger(self.name)
-        return logging.LoggerAdapter(logger, {"spider": self})
+        return SpiderLoggerAdapter(logger, {"spider": self})
```

Comment:
```suggestion
        return SpiderLoggerAdapter(logger, {"spider": self.name})
```
Maybe also change `self` to `self.name`? So we get actual spider name in logs instead of `<Spider 'spider_name' at 0x1b2fc089580>`

Reply:
> Probably not the best idea since it's gonna change current behavior

Yeah